frontend/dockerfile/builder/build.go

```diff
@@ -317,8 +318,14 @@ func Build(ctx context.Context, c client.Client) (*client.Result, error) {
 	}
 
 	if _, ok := opts["cmdline"]; !ok {
-		ref, cmdline, loc, ok := dockerfile2llb.DetectSyntax(bytes.NewBuffer(dtDockerfile))
-		if ok {
+		if cmdline, ok := opts[keySyntax]; ok {
+			p := strings.SplitN(strings.TrimSpace(cmdline), " ", 2)
```

Comment:
Why do we need to SplitN?

Reply:
I copy the same logic from `DetectSyntax` for compatibility. 
```Go
func DetectSyntax(r io.Reader) (string, string, []parser.Range, bool) {
    	p := strings.SplitN(v.Value, " ", 2)
	return p[0], v.Value, v.Location, true
```

Reply:
@tonistiigi PTAL?

Reply:
For performance.  As we only read the first element, splitting the rest is not needed.